src/backend/web/validation.js

```diff
@@ -89,7 +89,27 @@ const validateQuery = () => {
   };
 };
 
+// param validation starts here
+// paramValidation rules
+const validatePostsIdParamRules = () => {
+  return [param('id', 'Id Length is invalid').isLength({ min: 10, max: 10 })];
+};
+
+// paramValidationFunction
+const validatePostsIdParam = () => {
+  return async (req, res, next) => {
+    await Promise.all(validatePostsIdParamRules().map((rule) => rule.run(req)));
+
+    const errors = validationResult(req);
+    if (errors.isEmpty()) {
+      return next();
+    }
+    return res.status(400).json({ errors: errors.array() });
+  };
+};
+
```

Comment:
There is a common function in the current stage of validation.js 
`// creates a validation middleware with any given rules
const validate = (rules) => {
  return async (req, res, next) => {
    await Promise.all(rules.map((rule) => rule.run(req)));

    const result = validationResult(req);
    if (result.isEmpty()) {
      return next();
    }

    const errors = result.array();
    return res.status(400).send(errors);
  };
};`
So this function should be narrow to something similar to as we try to have dry code 
`module.exports = {
  validateNewFeed,	  
  validateQuery: () => validate(queryValidationRules),
  validatePostsQuery: () => validate(postsQueryValidationRules),
};`
Best way to do this is to have your code save somewhere, rebase your master start adding your code according to the current state of validation.js. You can also check this commit https://github.com/Seneca-CDOT/telescope/pull/1361/commits/1a2f541c28399c163b8d560189c2f66d2f481313

Reply:
@tonyvugithub I have a hard time to resolve commit conflict......I fixed it you mentioned but my commit is twist. I will update it to my master and wil pull request soon

Reply:
@tonyvugithub Finally I made 5 commit to single commit. Thanks!
@zg3d  I did my part by following your style.

so, Could you check my work?